File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/largepicture/LargePictureItemAdapter.java
package com.didichuxing.doraemonkit.kit.largepicture;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.TextView;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.kit.core.SettingItem;
import com.didichuxing.doraemonkit.widget.recyclerview.AbsRecyclerAdapter;
import com.didichuxing.doraemonkit.widget.recyclerview.AbsViewBinder;

/**
 * Created by jintai on 2019/9/24.
 */

public class LargePictureItemAdapter extends AbsRecyclerAdapter<AbsViewBinder<SettingItem>, SettingItem> {
    private OnSettingItemClickListener mOnSettingItemClickListener;
    private OnSettingItemSwitchListener mOnSettingItemSwitchListener;

    public LargePictureItemAdapter(Context context) {
        super(context);
    }

    @Override
    protected AbsViewBinder<SettingItem> createViewHolder(View view, int viewType) {
        return new SettingItemViewHolder(view);
    }

    @Override
    protected View createView(LayoutInflater inflater, ViewGroup parent, int viewType) {
        return inflater.inflate(R.layout.dk_item_setting, parent, false);
    }

    public class SettingItemViewHolder extends AbsViewBinder<SettingItem> {
        private TextView mDesc;
        private CheckBox mMenuSwitch;
        private ImageView mIcon;
        private TextView mRightDesc;

        public SettingItemViewHolder(View view) {
            super(view);
        }

        @Override
        protected void getViews() {
            mMenuSwitch = getView(R.id.menu_switch);
            mDesc = getView(R.id.desc);
            mIcon = getView(R.id.right_icon);
            mRightDesc = getView(R.id.right_desc);
        }

        @Override
        public void bind(final SettingItem settingItem) {
            mDesc.setText(settingItem.desc);
            if (settingItem.canCheck) {
                mMenuSwitch.setVisibility(View.VISIBLE);
                mMenuSwitch.setChecked(settingItem.isChecked);
                mMenuSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                    @Override
                    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                        settingItem.isChecked = isChecked;
                        mOnSettingItemSwitchListener.onSettingItemSwitch(mMenuSwitch, settingItem, isChecked);
                    }
                });
            }
            if (settingItem.icon != 0) {
                mIcon.setVisibility(View.VISIBLE);
                mIcon.setImageResource(settingItem.icon);
            }
            if (!TextUtils.isEmpty(settingItem.rightDesc)) {
                mRightDesc.setVisibility(View.VISIBLE);
                mRightDesc.setText(settingItem.rightDesc);
            }
        }

        @Override
        protected void onViewClick(View view, SettingItem data) {
            super.onViewClick(view, data);
            if (mOnSettingItemClickListener != null) {
                mOnSettingItemClickListener.onSettingItemClick(view, data);
            }
        }
    }

    public void setOnSettingItemClickListener(OnSettingItemClickListener onSettingItemClickListener) {
        mOnSettingItemClickListener = onSettingItemClickListener;
    }

    public void setOnSettingItemSwitchListener(OnSettingItemSwitchListener onSettingItemSwitchListener) {
        mOnSettingItemSwitchListener = onSettingItemSwitchListener;
    }

    public interface OnSettingItemClickListener {
        void onSettingItemClick(View view, SettingItem data);
    }

    public interface OnSettingItemSwitchListener {
        void onSettingItemSwitch(View view, SettingItem data, boolean on);
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/largepicture/LargePictureItemAdapter.kt
package com.didichuxing.doraemonkit.kit.largepicture

import android.text.TextUtils
import android.view.View
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.core.SettingItem
import com.didichuxing.doraemonkit.widget.bravh.BaseQuickAdapter
import com.didichuxing.doraemonkit.widget.bravh.viewholder.BaseViewHolder

/**
 * @author:  maple
 * @time:  2020/6/8 - 14:35
 * @desc: 大图配置列表
 */
class LargePictureItemAdapter()
    : BaseQuickAdapter<SettingItem, BaseViewHolder>(R.layout.dk_item_setting, mutableListOf()) {
    private var mOnSettingItemClickListener: LargePictureItemAdapter.OnSettingItemClickListener? = null
    private var mOnSettingItemSwitchListener: LargePictureItemAdapter.OnSettingItemSwitchListener? = null
    override fun convert(holder: BaseViewHolder, settingItem: SettingItem) {

        val mMenuSwitch = holder.getView<CheckBox>(R.id.menu_switch)
        val mDesc = holder.getView<TextView>(R.id.desc)
        val mIcon = holder.getView<ImageView>(R.id.right_icon)
        val mRightDesc = holder.getView<TextView>(R.id.right_desc)
        mDesc.setText(settingItem.desc)
        if (settingItem.canCheck) {
            mMenuSwitch.visibility = View.VISIBLE
            mMenuSwitch.isChecked = settingItem.isChecked
            mMenuSwitch.setOnCheckedChangeListener { buttonView, isChecked ->
                settingItem.isChecked = isChecked
                mOnSettingItemSwitchListener!!.onSettingItemSwitch(mMenuSwitch, settingItem, isChecked)
            }
        }
        if (settingItem.icon != 0) {
            mIcon.visibility = View.VISIBLE
            mIcon.setImageResource(settingItem.icon)
        }
        if (!TextUtils.isEmpty(settingItem.rightDesc)) {
            mRightDesc.visibility = View.VISIBLE
            mRightDesc.text = settingItem.rightDesc
        }
        holder.itemView.setOnClickListener { v->
            mOnSettingItemClickListener!!.onSettingItemClick(v, settingItem)
        }
    }

    fun setOnSettingItemClickListener(onSettingItemClickListener: OnSettingItemClickListener) {
        mOnSettingItemClickListener = onSettingItemClickListener
    }

    fun setOnSettingItemSwitchListener(onSettingItemSwitchListener: OnSettingItemSwitchListener) {
        mOnSettingItemSwitchListener = onSettingItemSwitchListener
    }

    interface OnSettingItemClickListener {
        fun onSettingItemClick(view: View?, data: SettingItem?)
    }

    interface OnSettingItemSwitchListener {
        fun onSettingItemSwitch(view: View?, data: SettingItem?, on: Boolean)
    }
}